week3/nametransformer.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+
+import nltk.stem
+
+class NameTransformer(object):
+    '''
+    Quick class to provide a method for transforming product names.
+    '''
+
+    def transform(self, name):
+        '''
+        Transforms a name, converting to lowercase, removing punctuation,
+        setmming, and so on.
+        '''
+        lowerCase = name.lower()
```

Comment:
I haven't tried it but I wonder if there is an option to make the NLTK stemmer automatically lowercase and remove some characters for us automatically. Might be worth checking out.